commit b611fddc0435738e64453bbf1dadd4b12a801858
Author: Christoph Hellwig <hch@lst.de>
Date:   Wed Sep 18 07:30:08 2024 +0200

    xfs: don't ifdef around the exact minlen allocations
    
    Exact minlen allocations only exist as an error injection tool for debug
    builds.  Currently this is implemented using ifdefs, which means the code
    isn't even compiled for non-XFS_DEBUG builds.  Enhance the compile test
    coverage by always building the code and use the compilers' dead code
    elimination to remove it from the generated binary instead.
    
    The only downside is that the alloc_minlen_only field is unconditionally
    added to struct xfs_alloc_args now, but by moving it around and packing
    it tightly this doesn't actually increase the size of the structure.
    
    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Reviewed-by: Darrick J. Wong <djwong@kernel.org>
    Signed-off-by: Carlos Maiolino <cem@kernel.org>

diff --git a/fs/xfs/libxfs/xfs_alloc.c b/fs/xfs/libxfs/xfs_alloc.c
index 59326f84f6a5..04f64cf9777e 100644
--- a/fs/xfs/libxfs/xfs_alloc.c
+++ b/fs/xfs/libxfs/xfs_alloc.c
@@ -2766,7 +2766,6 @@ xfs_alloc_commit_autoreap(
 		xfs_defer_item_unpause(tp, aarp->dfp);
 }
 
-#ifdef DEBUG
 /*
  * Check if an AGF has a free extent record whose length is equal to
  * args->minlen.
@@ -2806,7 +2805,6 @@ xfs_exact_minlen_extent_available(
 
 	return error;
 }
-#endif
 
 /*
  * Decide whether to use this allocation group for this allocation.
@@ -2880,15 +2878,14 @@ xfs_alloc_fix_freelist(
 	if (!xfs_alloc_space_available(args, need, alloc_flags))
 		goto out_agbp_relse;
 
-#ifdef DEBUG
-	if (args->alloc_minlen_only) {
+	if (IS_ENABLED(CONFIG_XFS_DEBUG) && args->alloc_minlen_only) {
 		int stat;
 
 		error = xfs_exact_minlen_extent_available(args, agbp, &stat);
 		if (error || !stat)
 			goto out_agbp_relse;
 	}
-#endif
+
 	/*
 	 * Make the freelist shorter if it's too long.
 	 *